doc/message-structure.md

```diff
@@ -0,0 +1,18 @@
+# Log message structure
+
+Within monolog log messages are passed around as arrays, for example to processors or handlers.
+The table below describes which keys are always available for every log message.
+
+key        | type                      | description
+-----------|---------------------------|-------------------------------------------------------------------------------
+message    | string                    | The log message. This string may contain placeholders that will be replaced by variables from the context, e.g., "User %username% logged in" with `['username' => 'John']` as context will be written as "User John logged in".
+level      | int                       | Severity of the log message. See log levels described in [01-usage.md](01-usage.md).
+level_name | string                    | String representation of log level.
+context    | array                     | Arbitrary data passed with the constuction of the message. For example the username of the current user or their IP addres.
+channel    | string                    | The channel this message was logged to.
+datetime   | Monolog\DateTimeImmutable | Date and time when the message was logged. Class extends `\DateTimeImmutable`.
+extra      | array                     | A placeholder array where processors can put additional data. Always available, but empty if there are no processors registered.
+
+At first glance `context` and `extra` look very similar, and they are in the sense that they both carry arbitrary data that is related to log the log message somehow.
+The main difference is that `context` can be supplied in user land (it is the 3rd parameter to `Logger::addRecord()`) whereas `extra` is internal only and can be filled processors.
```

Comment:
I think the end of the sentence here should be something like "can be filled *by* processors`.